api_unit_test.go

```diff
@@ -77,25 +77,6 @@ func TestGetReaderSize(t *testing.T) {
 		t.Fatalf("Reader length doesn't match got: %v, want: %v", size, len("Hello World"))
 	}
 
-	// Create request channel.
-	reqCh := make(chan readRequest)
```

Comment:
Do not remove this test. 

Send back 'objInfo' through a channel. and still proceed to execute getReaderSize() this would in-fact tell if the Size is appropriately set.